generic-helpers/src/main/java/io/elimu/a2d2/helpers/MixPanelService.java

```diff
@@ -0,0 +1,72 @@
+package io.elimu.a2d2.helpers;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.drools.core.spi.KnowledgeHelper;
+
+import io.elimu.a2d2.exception.WorkItemHandlerException;
+import io.elimu.a2d2.genericmodel.ServiceRequest;
+
+public class MixPanelService {
+	private static final MixPanelService INSTANCE = new MixPanelService();
+
+	private ThreadLocal<String> distinctId = ThreadLocal.withInitial(() -> "");
```

Comment:
Why would the `distinctId` be thread local? It will change when processing any event for a different patient.

Reply:
This class is a singleton, so the thread local is to guarantee two service calls done at the same time, from different threads, will not share the identity